include/eeprom/core/hardware_iface.hpp

```diff
@@ -3,12 +3,28 @@
 namespace eeprom {
 namespace hardware_iface {
 
+enum EEPromAddressType {
```

Comment:
nit: if we make these `enum class` we don't have to have c-style NAMESPACE_PREFIX_ELEMENT member names

Reply:
made the other changes and I think I did what you meant here. I also cleaned up a bunch of extra namespace things we didn't need.